### ISSUE TITLE
SpectrumTrack extends SliderTrack is odd.

### ISSUE THREAD

**Issue by samreid** [phetsims/scenery-phet#505]
From #498 

> SpectrumTrack extends SliderTrack is odd. It's a track on top of a track. How about factoring out a base class, something like:

> SliderTrack - abstract, needs a Node for the track
   DefaultSliderTrack extends SliderTrack - provides a Rectangle for the track 
   SpectrumSlider extends SliderTrack - provides a SpectrumNode for the track

I replied:

That sounds nice, I'm going to upgrade SliderTrack to ES6/class and look into it further.

Port to ES6 was done in https://github.com/phetsims/sun/commit/563b9202e7a9d632e29799d0653d36bb14a8f6da

**Comment by samreid** [comment]
Proposed fixes committed, please review.

**Comment by pixelzoom** [comment]
I did a quickie review because otherwise I won't be able to get to this until July.   Looks good, makes sense.  

Not exactly specific to this issue, but I noticed it now that there's a "track" hierarchy...  Consider renaming `SpectrumTrack` and `SpectrumThumb` to `SpectrumSliderTrack` and `SpectrumSliderThumb` respectively.   They are parts of the SpectrumSlider, and their current names are missing a bit of information about what they are related to.   This would also match naming of Slider, SliderTrack, DefaultSliderTrack, SliderThumb.
